src/js/user.js

```diff
@@ -3,11 +3,36 @@
 var _ = require('lodash'),
     Gogs = require('gogs-client');
 
-function UserManager(auth, server) {
-
+function UserManager(auth, server) {   
+    
     var api = new Gogs(server + '/api/v1'),
         tokenStub = {name: 'btt-writer-desktop'};
 
+    const MAX_PAGE_SIZE = 50
+
+    const fetchRepoExhaustively = async function (uid, query, limit) {
+        limit = limit || MAX_PAGE_SIZE;
+        let page = 1;
+        let repoList = [];
+        let hasMore = true;
+
+        while (hasMore) {
```

Comment:
This is probably just my lack of domain knowledge on this problem, but what guarantees that this while loop won't run forever?

Reply:
look at line 29

Reply:
I see line 29. My question is how is it guaranteed that line 29 will ever be reached?

Reply:
Oh it's because the page is incremented every time. And there's only a finite number of pages. I see now. Line 27 answers my question.